pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,319 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository that implements an in-memory repository
+based on dictionaries
+
+For the module-level documentation, see mocksupport.rst.
+"""
+
+import six
+
+from pywbem._nocasedict import NocaseDict
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+from pywbem._utils import _format
+from ._baserepository import BaseObjStore, BaseRepository
+
+
+class InMemoryObjStore(BaseObjStore):
+    """
+    """
+    def __init__(self, case_insensitive_names, cim_object_type, repository):
+        """
+
+          Parameters:
+
+            case_insensitive_names(:class:`py:bool`):
+                Defines whether the data store for this type handles
+                case insesitive names or not
+
+            cim_object_type (TODO):
+             CIMType of the object storded
+
+            repository (TODO)
+                handle of the repository for the type defined in cim_object_Type
+                within a single namespace
```

Comment:
On the repository argument:
* I don't understand from the description what its type is or how it works.
* It seems to establish a somewhat unhealthy backlink to the repository. I think we should be able to define an object store for a single CIM type without having to know which repository this store will be used in.

Reply:
Obsolete Rewritten in previous commit